presto-elasticsearch/src/test/java/com/facebook/presto/elasticsearch/EmbeddedElasticsearchNode.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.elasticsearch;
+
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.transport.Netty4Plugin;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+
+public class EmbeddedElasticsearchNode
+        implements Closeable
+{
+    private final File elasticsearchDirectory;
+    private final ElasticsearchNode node;
+
+    private final AtomicBoolean started = new AtomicBoolean();
```

Comment:
I think Instead of these two booleans you can have

```
private final AtomicBoolean running = new AtomicBoolean();
```

Then `start()` will be 

```
if (running.compareAndSet(false, true)) {
    node.start();
}
```

and `close()` will be 

```
if (running.compareAndSet(true, false)) {
    node.close();
    deleteRecursively(elasticsearchDirectory.toPath(), ALLOW_INSECURE);
}
```